src/server.cpp

```diff
@@ -411,6 +408,52 @@ static int statusToExitCode(const Status& status) {
     return OVMS_EX_FAILURE;
 }
 
+std::variant<std::pair<ServerSettingsImpl, ModelsSettingsImpl>, std::pair<int, std::string>> Server::parseArgs(int argc, char** argv) {
+    try {
+        CLIParser parser;
+        ServerSettingsImpl serverSettings;
+        ModelsSettingsImpl modelsSettings;
+        auto successOrExit = parser.parse(argc, argv);
+        // Check for error in parsing
+        if (!std::holds_alternative<bool>(successOrExit)) {
+            auto printAndExit = std::get<std::pair<int, std::string>>(successOrExit);
+            return printAndExit;
+        }
+        parser.prepare(&serverSettings, &modelsSettings);
+        return std::make_pair(serverSettings, modelsSettings);
+    } catch (const std::exception& e) {
+        return std::make_pair(OVMS_EX_USAGE, e.what());
+    }
+}
+
+int Server::startServerFromSettings(ServerSettingsImpl& serverSettings, ModelsSettingsImpl& modelsSettings) {
+    installSignalHandlers();
+    int result = OVMS_EX_OK;
+
+    try {
+        Status ret = startFromSettings(&serverSettings, &modelsSettings);
+        ModulesShutdownGuard shutdownGuard(*this);
+        if (!ret.ok()) {
+            return statusToExitCode(ret);
+        }
+        while (!shutdown_request &&
+               (serverSettings.serverMode == HF_PULL_AND_START_MODE || serverSettings.serverMode == SERVING_MODELS_MODE)) {
+            std::this_thread::sleep_for(std::chrono::milliseconds(200));
+        }
+        if (shutdown_request == 2) {
+            SPDLOG_ERROR("Illegal operation. OVMS started on unsupported device");
+        }
+    } catch (const std::exception& e) {
+        SPDLOG_ERROR("Exception; {}", e.what());
+        result = OVMS_EX_FAILURE;
+        shutdown_request = 3;
```

Comment:
Separate variable - for now it serves as comunnication from user to ovms to request stop. It would be better to have sth else to signal the other way arround -> that ovms stopped and ovms wrapper (service) needs to know. Ensure its reset when ovms is starting